### PR TITLE
Pull request: SciScore additional analysis

### PR DESCRIPTION
**Pull request by elisabascunan**
For SciScore analysis, there are two new tables with:
- total number of ctgov IDs found in methods section
- total number of ctgov IDs found by ctgov that were in methods section.

Additionally, the titles of the location ID per tool plots were modified, elimintating the phrase "_detected by toolname_"

### CODE REVIEW

**Comment by delwen** [comment]
@elisabascunan see below some additional changes needed to the dataset. Please review the corrections again before incorporating them as a sanity check. I think the easiest is if you make these changes in the Google Sheet, and then read in the updated dataset in your analysis workflow. That way the changes will be integrated everywhere (but please push back on this if you're seeing an issue I overlooked!).

1. Rename "TrialIdentifier" to "TRNscreener" in line with the updates in the main paper.

2. NCT0002523: when I looked into some of the IDs/papers that were not detected by SciScore, I saw that this ID is in the results section of the paper (and not in the methods as indicated in the extraction form), so we should correct this in the extraction form and analysis.

3. NCT00418041: this was also detected in the notes/metadata of the article. We could potentially reflect this by setting `id_in_other_location` to TRUE (+ comment), unless you didn't typically consider metadata?

4. NCT01768585: right now, we have two entries for this trial ID (NCT01768585 and NCT 01768585) for the same paper (PMC8024658). We decided that we will collapse these, since we only have a second row for this ID because ctregistries is cleaning the detected identifier. In reality, the ID as written in the paper was found by TrialIdentifier (or TRNscreener), ctregistries, and NCT naive tool. So concretely, you can merge these two rows and keep as Identifier "NCT01768585" and set `sciscore_hit` to `FALSE`, and `trialidentifier_hit` (or trnscreener_hit), `ctregistries_hit`, and `nct_hit` to `TRUE`. Ideally add a comment explaining that these two entries were merged so we can keep track. Note that this resolves one of the questions we had in this issue: https://github.com/delwen/screenit-tool-comparison/issues/2#issuecomment-1976881390.

5. Once these changes are made, re-generate the analysis.

Thanks, and let me know if this raises any questions!

**Comment by elisabascunan** [comment]
@delwen thank you for the feedback! I made the changes on the dataset and updated the file regset.csv accordingly. I also commited the newest version of the quarto report with the correct numbers. Let me know if you need me to add anything else.

File: descriptive-analysis.qmd [hunk 1]
```diff
@@ -6,19 +6,19 @@ editor: visual
 
 ## Introduction
 
-This document presents the descriptive results from the analysis of 4 different tools (`sciscore`, `trialidentifier`, `ctregistries` and `nct`). The aim of these tools is to screen papers for one or more registration IDs (protocol, trial registry, etc).
+This document presents the descriptive results from the analysis of 4 different tools (`sciscore`, `TRNscreener`, `ctregistries` and `nct`). The aim of these tools is to screen papers for one or more registration IDs (protocol, trial registry, etc).
 
 #### Data description
 
-This analysis was limited to papers (n = 117) where a hit was found by one or more tools. Overall, 200 IDs were analyzed. These IDs were grouped in a data file named `regset.csv`, with the following variables/columns:
+This analysis was limited to papers (n = 117) where a hit was found by one or more tools. Overall, 199 IDs were analyzed. These IDs were grouped in a data file named `regset.csv`, with the following variables/columns:
```
Comment (delwen) [inline descriptive-analysis.qmd:13]: More exactly, a total of 199 unique registration identifiers _unique per paper_ (as described in manuscript)

```
# number of unique identifiers in the set
data |> distinct(Identifier) |> nrow()
196

# number of unique identifiers AND paper in the set
data |> distinct(Identifier, PMCID) |> nrow()
199
```

File: descriptive-analysis.qmd [hunk 2]
```diff
@@ -199,7 +199,7 @@ wordcloud(words = other_location$locations,
 
 ### NOTE:
 
-The IDs analyzed could be in **one or more locations** of the paper, that is why there are more cases out of a total of 200 IDs. This is important to remember for a following section that analyzes the location of the ID by tool. Some tools searched for the IDs in a particular section of a paper, but if the ID was also present in another section of the same paper, it was also recorded on the datafile.
+The IDs analyzed could be in **one or more locations** of the paper, that is why there are more cases out of a total of 199 IDs. This is important to remember for a following section that analyzes the location of the ID by tool. Some tools searched for the IDs in a particular section of a paper, but if the ID was also present in another section of the same paper, it was also recorded on the datafile.
```
Comment (delwen) [inline descriptive-analysis.qmd:202]: _" Some tools searched for the IDs in a particular section of a paper, but if the ID was also present in another section of the same paper, it was also recorded on the datafile."_

Upon re-reading, I find this a bit confusing. I would maybe rephrase to say that SciScore was _run_ on the methods sections (none of the tools are designed to detect a particular section of a paper, SciScore was just run on the methods).

_"...but if the ID was also present in another section of the same paper, it was also recorded on the datafile."_: you mean that all locations of an ID in a given paper were recorded, right?

File: descriptive-analysis.qmd [hunk 3]
```diff
@@ -291,33 +291,56 @@ text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex
```
Comment (delwen) [inline descriptive-analysis.qmd:291]: _"`Sciscore` is designed to search the IDs in a specific section of the paper (methods)."_

Not strictly correct. SciScore was run on the methods sections.

File: descriptive-analysis.qmd [hunk 4]
```diff
@@ -291,33 +291,56 @@ text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex
 
 ```{r message = FALSE, warning = FALSE}
 
-sciscore <- data %>% filter(sciscore_hit)
-sciscore %>% count(id_in_abstract, id_in_methods, id_in_other_location)
+sciscore2 <- data %>% filter(sciscore_hit)
+sciscore2 %>% count(id_in_abstract, id_in_methods, id_in_other_location)
```
Comment (delwen) [inline descriptive-analysis.qmd:295]: Why `sciscore2`? I think you can just continue using the variable `sciscore` above, right?

Also, add `kable()` for a cleaner table

File: descriptive-analysis.qmd [hunk 5]
```diff
@@ -291,33 +291,56 @@ text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex
...
-### Tool 2: trialidentifier
+Additionally, we compared the total number of ctgov IDs present in the methods section, to the ctgovs IDs present in methods section were sciscore_hit was TRUE, to see if the tool missed any of these IDs in this particular section:
 
 ```{r message = FALSE, warning = FALSE}
-#subset of trialidentifier_hit = TRUE
-trialidentifier <- subset(data, data$trialidentifier_hit %in% c("TRUE"))
 
-trialidentifier %>%
+#dataset with only methods section = TRUE
+methods_ids <- subset(data, data$id_in_methods %in% c("TRUE"))
```
Comment (delwen) [inline descriptive-analysis.qmd:307]: you could also do:

`methods_ids <- data |> filter(id_in_methods)`

(can use `filter` and `id_in_methods` is bool)

File: descriptive-analysis.qmd [hunk 6]
```diff
@@ -291,33 +291,56 @@ text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex
...
+trnscreener_hit %>%
   count(id_type_group, id_type) %>%
   kable()
 ```
 
-`trialidentifier` identified 139 trial registry number IDs, 136 from ctgov and 3 from isrctn.
+`TRNscreener` identified 139 trial registry number IDs, 136 from ctgov and 3 from isrctn.
 
-The IDs were present in more than 2 locations on the paper: abstract (35 cases), methods (65) and other location (80):
+The IDs were present in more than 2 locations on the paper: abstract (35 cases), methods (64) and other location (82):
```
Comment (delwen) [inline descriptive-analysis.qmd:338]: minor correction: 'in' the paper

File: descriptive-analysis.qmd [hunk 7]
```diff
@@ -376,7 +399,7 @@ ctregistries %>%
   kable()
 ```
 
-Finally, the IDs were present in more than 2 locations on the paper: abstract (43 cases), methods (91), other location (112):
+Finally, the IDs were present in more than 2 locations on the paper: abstract (43 cases), methods (91), other location (113):
```
Comment (delwen) [inline descriptive-analysis.qmd:402]: 'in' the paper

I also wanted to comment on another point in the code above, but was not able to enter the comment. This bit of code repeats and I think can just be shown once, wherever you think is most relevant:

```
ctregistries %>%
  filter(id_type_group == "trn") %>%
  count(id_type) %>%
  arrange(desc(n)) %>%
  kable()
```

File: descriptive-analysis.qmd [hunk 8]
```diff
@@ -412,9 +435,9 @@ nct %>%
   kable()
 ```
 
-`nct` identified 137 trial registry number IDs, all from ctgov.
+`nct` identified 136 trial registry number IDs, all from ctgov.
 
-The IDs were present in more than 2 locations on the paper: abstract (34 cases), methods (63), other location (80):
+The IDs were present in more than 2 locations on the paper: abstract (34 cases), methods (61), other location (82):
```
Comment (delwen) [inline descriptive-analysis.qmd:440]: 'in' the paper

File: descriptive-analysis.qmd [hunk 9]
```diff
@@ -6,19 +6,19 @@ editor: visual
 
 ## Introduction
 
-This document presents the descriptive results from the analysis of 4 different tools (`sciscore`, `trialidentifier`, `ctregistries` and `nct`). The aim of these tools is to screen papers for one or more registration IDs (protocol, trial registry, etc).
+This document presents the descriptive results from the analysis of 4 different tools (`sciscore`, `TRNscreener`, `ctregistries` and `nct`). The aim of these tools is to screen papers for one or more registration IDs (protocol, trial registry, etc).
```
Comment (delwen) [inline descriptive-analysis.qmd:9]: _"The aim of these tools is to screen papers for one or more registration IDs (protocol, trial registry, etc)."_

more exactly, the aim of these tools is to detect registration identifiers (protocol, trial registration, etc.). The tools are mostly regexes that are run on any text

File: descriptive-analysis.qmd [hunk 10]
```diff
@@ -6,19 +6,19 @@ editor: visual
...
 
 #### Data description
 
-This analysis was limited to papers (n = 117) where a hit was found by one or more tools. Overall, 200 IDs were analyzed. These IDs were grouped in a data file named `regset.csv`, with the following variables/columns:
+This analysis was limited to papers (n = 117) where a hit was found by one or more tools. Overall, 199 IDs were analyzed. These IDs were grouped in a data file named `regset.csv`, with the following variables/columns:
 
 -   **Identifier**: character. It contains the ID detected. Example: `NCT03192410`.
 
 -   **PMCID**: character. PMC identifier of the paper where the ID was found.
```
Comment (delwen) [inline descriptive-analysis.qmd:17]: PubMed Central (PMC) identifier

File: descriptive-analysis.qmd [hunk 11]
```diff
@@ -178,7 +178,7 @@ bar_midpoints <- barplot(count_true, plot = FALSE)
 text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex = 0.8)
 ```
 
-**What were the other locations?** - discussion (53 cases), introduction (23 cases), acknowledgements (16 cases ), results (11 cases), etc.
+**What were the other locations?** - discussion (53 cases), introduction (23 cases), acknowledgements (16 cases ), results (12 cases), etc.
```
Comment (delwen) [inline descriptive-analysis.qmd:181]: instead of 'etc.', I suggest to give all possible locations found

File: descriptive-analysis.qmd [hunk 12]
```diff
@@ -224,7 +224,7 @@ data %>%
   kable()
 ```
 
-Out of 117 papers reviewed, 112 were research articles, and were associated with 159 IDs. The remaining 5 articles in which 41 IDs were found appeared to be either protocol papers, review of findings and an article that did not present a methods section.
+Out of 117 papers reviewed, 112 were research articles, and were associated with 158 IDs. The remaining 5 articles in which 41 IDs were found appeared to be either protocol papers, review of findings and an article that did not present a methods section.
```
Comment (delwen) [inline descriptive-analysis.qmd:227]: When you look at the 112 research articles and associate them with 158 IDs: you still have 3 duplicate Identifiers in there (i.e., in 3 cases, the same Identifier was found in >1 paper, which is fine). But for this analysis you might consider reporting the number of _unique_ identifiers associated with the 112 research articles. Thoughts?

File: descriptive-analysis.qmd [hunk 13]
```diff
@@ -238,7 +238,7 @@ data %>%
   kable()
 ```
 
-Overall, the tools agreed in 54 cases, and disagreed in 146. Additionally, they only agreed when the ID was a trn:
+Overall, the tools agreed in 54 cases, and disagreed in 145. Additionally, they only agreed when the ID was a trn:
```
Comment (delwen) [inline descriptive-analysis.qmd:241]: it might be worth noting here again that this is partly expected given that SciScore was only run on the methods section of papers (remind the reader of important points to consider when reflecting on the findings)

File: descriptive-analysis.qmd [hunk 14]
```diff
@@ -291,33 +291,56 @@ text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex
...
+sciscore2 <- data %>% filter(sciscore_hit)
+sciscore2 %>% count(id_in_abstract, id_in_methods, id_in_other_location)
 ```
 
 Although the majority of IDs were in both methods and another other location of the paper, there were 2 cases where the IDs were found in the introduction and not in the methods section: `NCT03349060` and `NCT03575871`.
 
 These two IDs were used as a reference in the same paper ([click here to see research article](https://pubmed.ncbi.nlm.nih.gov/34406366/)).
 
-### Tool 2: trialidentifier
+Additionally, we compared the total number of ctgov IDs present in the methods section, to the ctgovs IDs present in methods section were sciscore_hit was TRUE, to see if the tool missed any of these IDs in this particular section:
```
Comment (delwen) [inline descriptive-analysis.qmd:302]: _where_ sciscore_hit was TRUE

File: descriptive-analysis.qmd [hunk 15]
```diff
@@ -291,33 +291,56 @@ text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex
...
+
+#table sciscore - ctgov in methods
+methods_ids %>% 
+  count(id_type, sciscore_hit) %>%
+  arrange(desc(n)) %>%
+  kable()
+
+```
+
+Out of 63 ctgov IDs present in the methods section, `Sciscore` found only 52 ctgov IDs in the method section.
```
Comment (delwen) [inline descriptive-analysis.qmd:323]: Should it not be 61 instead of 63, as also indicated in the tables you generate? Also keep in mind these are 61 unique identifier-PMCID combinations (but 60 unique identifiers).

A general comment for the analysis above: since you say you focus on ctgov, you could perhaps filter for ctgov from the start? I think this would make it a bit easier to read. Something like this:

```
# filter for ctgov and ID present in methods
ctgov_methods <- data |> filter(id_type == "ctgov", id_in_methods)

***
SIDE NOTE:
# number of unique Identifier-PMCID combinations in this subet
ctgov_methods |> distinct(Identifier, PMCID) |> nrow()
61

# number of unique Identifiers in this subset
ctgov_methods |> distinct(Identifier) |> nrow()
60
***

# show how many were found by sciscore
ctgov_methods |> count(sciscore_hit)
```

File: descriptive-analysis.qmd [hunk 16]
```diff
@@ -444,18 +467,18 @@ text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex
 
 data %>%
   filter(id_type_group == "trn") %>% 
-  count(ctregistries_hit, sciscore_hit, trialidentifier_hit, nct_hit) %>%
+  count(ctregistries_hit, sciscore_hit, trnscreener_hit, nct_hit) %>%
   kable()
```
Comment (delwen) [inline descriptive-analysis.qmd:471]: since you filter for `id_type_group == "trn"`, I suggest to add in the title of this section that it focuses on trial registration numbers

File: descriptive-analysis.qmd [hunk 17]
```diff
@@ -444,18 +467,18 @@ text(bar_midpoints, count_true, labels = count_true, pos = 3, col = "black", cex
...
--   **ctregistries** detected 28 trial registration ids that no other tool found, and only missed 3 ids that were detected by both trialidentifier and nct.
+-   **ctregistries** detected 28 trial registration ids that no other tool found, and only missed 2 ids that were detected by both TRNscreener and nct.
 
--   **sciscore** detected 54 ids that all the other tools also found, and it found only one extra id that was also detected by ctregistries. Overall the tool missed 114 trial registration ids.
+-   **sciscore** detected 54 ids that all the other tools also found, and it found only one extra id that was also detected by ctregistries. Overall the tool missed 113 trial registration ids.
 
--   **trialidentifier** seem to agree in the majority of cases with nct, but it found 3 additional ids. Overall the tool missed 30 trial registration ids.
+-   **TRNscreener** seem to agree in the majority of cases with nct, but it found 3 additional ids. Overall the tool missed 29 trial registration ids.
 
 -   **nct** found only one additional ids compared to trial identifier. Overall the tool missed 32 trial registration ids.
```
Comment (delwen) [inline descriptive-analysis.qmd:483]: need to replace `trial identifier`

Is this last statement still correct? I'm not seeing a case where nct found an identifier that TRNscreener did not find

Comment (delwen) [inline descriptive-analysis.qmd:483]: since the numbers are entered manually, could you double check this section again as it's one of the core analyses of the report? Thanks!

File: descriptive-analysis.qmd [hunk 18]
```diff
@@ -517,7 +540,7 @@ Funding IDs that were present in the negative control check, and therefore not d
```
Comment (delwen) [inline descriptive-analysis.qmd:540]: again a comment I couldn't enter in the right place: 

_"In this analysis, 71 of 137 ctgov trn's were used as reference."_: according to an earlier table, you have 136 ctgov identifiers. Please check the numbers again or generate them by integrating code in the text

File: descriptive-analysis.qmd [hunk 19]
```diff
@@ -517,7 +540,7 @@ Funding IDs that were present in the negative control check, and therefore not d
 
 ### TRN with missing digits
 
-There were two cases of ctgov IDs reported with a missing digit, but the tools `trialidentifier` and `nct` were still able to identify them.
+There were two cases of ctgov IDs reported with a missing digit, but the tools `TRNscreener` and `nct` were still able to identify them.
```
Comment (delwen) [inline descriptive-analysis.qmd:543]: wasn't there a case where ctregistries didn't a TRN with a missing digit? (maybe NCT2927821 or NCT0002523)? Might be an interesting observation to add?

**Review by delwen — CHANGES_REQUESTED** [review]
Thanks @elisabascunan! See feedback in comments, and some more general points below:

1. **NCT00418041 (metadata):** Thanks for making the requested changes to the regset dataset. For "NCT00418041" where I had suggested to flag that it was noted in the article metadata: did you typically encode this in other cases? It occurred to me that this might not have been the case if you were always checking the PDF. We should aim to be consistent either way.
2. **Numbers in Quarto:** Could you carefully check again any number that isn’t generated with code in the Quarto (see some discrepancies in specific comments)? I generally recommend to use inline code to automatically update all numbers in the report without having to check them manually (see https://quarto.org/docs/computations/inline-code.html).
3. **HTML:** After making the requested changes, could you regenerate the HTML?
4. **Name of NCT tool:** Note that we may need to change the name of the NCT tool, but let's hold off until naming is decided in the manuscript.

File: descriptive-analysis.qmd (hunk 1 above)
Comment (elisabascunan) [reply descriptive-analysis.qmd:13]: Added the text. However, what exactly do you mean by "_unique per paper_"? 
I want to clarify it in a note so that the reader knows what it is.

File: descriptive-analysis.qmd (hunk 2 above)
Comment (elisabascunan) [reply descriptive-analysis.qmd:202]: I agree it is quite confusing. I meant that even though we analyzed a total of 199 trn, when we look at the location analysis, the ID could be in one or more locations and therefore the numbers on those graphs will not add up to 199 IDs. I will rewrite the text to make it clearer. 

I think the last line should be deleted, and to add the clarification of SciScore (run only on the methods) on the section flagged below.

File: descriptive-analysis.qmd (hunk 3 above)
Comment (elisabascunan) [reply descriptive-analysis.qmd:291]: Updated the text to clarify this point.

File: descriptive-analysis.qmd (hunk 4 above)
Comment (elisabascunan) [reply descriptive-analysis.qmd:295]: Replaced it with using only `sciscore` and adding `kable()`

File: descriptive-analysis.qmd (hunk 6 above)
Comment (elisabascunan) [reply descriptive-analysis.qmd:338]: I searched all the "_on the paper_" of this section and replaced them "_with in the paper_".

File: descriptive-analysis.qmd (hunk 1 above)
Comment (delwen) [reply descriptive-analysis.qmd:13]: some IDs were found across >1 paper, so we need to make it clear to the reader that we mean 199 distinct ID-paper combinations. The alternative would be to say that 196 unique IDs were analysed, but since the paper level is relevant for the analysis (since we also look into sections of the paper were detected), I lean towards sticking with the phrasing above. Happy to consider other ways to put this across! Does that clarify?

File: descriptive-analysis.qmd (hunk 17 above)
Comment (elisabascunan) [reply descriptive-analysis.qmd:483]: Updated the analysis and the name of TRNscreener

Comment (elisabascunan) [reply descriptive-analysis.qmd:483]: Reviewed and double checked. Updated only `nct`

File: descriptive-analysis.qmd (hunk 19 above)
Comment (elisabascunan) [reply descriptive-analysis.qmd:543]: I looked it up and tthe two ctgov IDs with missing digits were not found by sciscore in one case, and by sciscore and ctregistries in the other case. Not sure if that adds much information for the report.

File: descriptive-analysis.qmd (hunk 12 above)
Comment (elisabascunan) [reply descriptive-analysis.qmd:227]: I think given the statements it does make more sense to analyze unique identifiers and not id-paper combinations. I will give it a go and see how it looks.

**Comment by elisabascunan** [comment]
HI @delwen 

I just went through most of the comments and I only have my doubts with the last one regarding shorter NCTs (stated above as a reply). To answer the main questions of this review:

- **NCT00418041 (metadata)**: I only checked the pdfs and I wanted to clarify what you mean by metadata. In this particular case, I thought I only missed it on the file as it is in the first page (see attached photo). If that is a section that you considered as metadata, then adding this category would  be consistent with the process. However, if it is something entirely different and not available from a pdf, then we might need to change this location category. 

Screenshot 2024-05-30 at 09 26 07

- **Numbers in Quarto**: I completed the manual review, but I will look into the option of replacing the numbers with inline code. 

I will wait for your comments before pushing the new version. Thank you in advance!
